eva-pipeline/src/test/java/embl/ebi/variation/eva/pipeline/steps/VariantsAnnotGenerateInputBatchTest.java

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright 2015-2016 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package embl.ebi.variation.eva.pipeline.steps;
+
+import com.mongodb.*;
+import com.mongodb.util.JSON;
+import embl.ebi.variation.eva.pipeline.annotation.generateInput.VariantAnnotationItemProcessor;
+import embl.ebi.variation.eva.pipeline.annotation.generateInput.VariantWrapper;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.opencb.opencga.storage.mongodb.variant.DBObjectToVariantConverter;
+import org.springframework.batch.item.ExecutionContext;
+import org.springframework.batch.item.ItemProcessor;
+import org.springframework.batch.item.data.MongoItemReader;
+import org.springframework.batch.item.file.FlatFileItemWriter;
+import org.springframework.batch.test.MetaDataInstanceFactory;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.core.io.support.PropertiesLoaderUtils;
+import org.springframework.data.mongodb.core.MongoTemplate;
+
+import java.io.*;
+import java.util.Collections;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertFalse;
+import static junit.framework.TestCase.assertTrue;
+
+/**
+ * @author Diego Poggioli
+ *
+ * Test {@link VariantsAnnotGenerateInputBatch}
+ */
+public class VariantsAnnotGenerateInputBatchTest {
+
+    private MongoClient mongoClient;
+    private String databaseName;
+    private String collectionName = "dummy";
+
+    private VariantsAnnotGenerateInputBatch variantsAnnotGenerateInputBatch;
+    private ExecutionContext executionContext;
+
+    // temporary output file
+    private File outputFile;
+
+    // reads the output file to check the result
+    private BufferedReader reader;
+
+    @Before
+    public void setUp() throws Exception {
+        databaseName = PropertiesLoaderUtils.loadAllProperties("application.properties").getProperty("mongo.db");
```

Comment:
Could the property for the database name be the same as in the rest of the app? ("dbName" right now)